underfs/oss/src/main/java/tachyon/underfs/oss/OSSUnderFileSystem.java

```diff
@@ -0,0 +1,593 @@
+    return key + FOLDER_SUFFIX;
+  }
+
+  /**
+   * Copies an object to another key.
+   * @param src the source key to copy
+   * @param dst the destination key to copy to
+   * @return true if the operation was successful, false otherwise
+   */
+  private boolean copy(String src, String dst) {
+    try {
+      src = stripPrefixIfPresent(src);
+      dst = stripPrefixIfPresent(dst);
+      LOG.info("Copying {} to {}", src, dst);
+      mOssClient.copyObject(mBucketName, src, mBucketName, dst);
+      return true;
+    } catch (ServiceException se) {
+      LOG.error("Failed to rename file {} to {}", src, dst);
+      return false;
+    }
+  }
+
+  /**
+   * Internal function to delete a key in OSS
+   * @param key the key to delete
+   * @return true if successful, false if an exception is thrown
+   */
+  private boolean deleteInternal(String key) {
+    try {
+      if (isFolder(key)) {
+        String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
+        mOssClient.deleteObject(mBucketName, keyAsFolder);
+      } else {
+        mOssClient.deleteObject(mBucketName, stripPrefixIfPresent(key));
+      }
+    } catch (ServiceException se) {
+      LOG.error("Failed to delete {}", key, se);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Gets the {@link StorageObject} representing the metadata of a key. If the key does not exist as
+   * a file or folder, null is returned
+   * @param key the key to get the object details of
+   * @return {@link StorageObject} of the key, or null if the key does not exist as a file or folder
+   */
+  private ObjectMetadata getObjectDetails(String key) {
+    try {
+      if (isFolder(key)) {
+        String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
+        return mOssClient.getObjectMetadata(mBucketName, keyAsFolder);
+      } else {
+        return mOssClient.getObjectMetadata(mBucketName, stripPrefixIfPresent(key));
+      }
+    } catch (ServiceException e) {
+      LOG.warn("Failed to get Object {}, return null", key);
+      return null;
+    }
+
+  }
+
+  /**
+   * Gets the parent key of the input key, or null if no parent exists
+   * @param key the key to get the parent of
+   * @return the the parent key
+   */
+  private String getParentKey(String key) {
+    // Root does not have a parent.
+    if (isRoot(key)) {
+      return null;
+    }
+    int separatorIndex = key.lastIndexOf(PATH_SEPARATOR);
+    if (separatorIndex < 0) {
+      return null;
+    }
+    return key.substring(0, separatorIndex);
+  }
+
+  /**
+   * Sets the OSS Client Configuration
+   * @param tachyonConf The TachyonConf
+   * @return The {@link ClientConfiguration} hold the OSS Client config
+   */
+  private ClientConfiguration initializeOSSClientConfig(TachyonConf tachyonConf) {
+    ClientConfiguration ossClientConf = new ClientConfiguration();
+    ossClientConf.setConnectionTimeout(
+        tachyonConf.getInt("tachyon.underfs.oss.connection.timeout.ms"));
```

Comment:
also please make sure `setConnectionTimeout` is also expecting a time in unit of ms?